python/cuml/cuml/linear_model/base.pyx

```diff
@@ -74,19 +74,34 @@ class LinearPredictMixin:
                 "Please fit the model first."
             )
         self.dtype = self.coef_.dtype
-        if len(self.coef_.shape) == 2 and self.coef_.shape[0] > 1:
-            # Handle multi-target prediction in Python.
+        multi_target = (len(self.coef_.shape) == 2 and self.coef_.shape[0] > 1)
+        intercept_is_scalar = isinstance(self.intercept_, (int, float, np.number))
+        if multi_target or not intercept_is_scalar:
             coef_arr = CumlArray.from_input(self.coef_).to_output('array')
             X_arr = CumlArray.from_input(
                 X,
                 check_dtype=self.dtype,
                 convert_to_dtype=(self.dtype if convert_dtype else None),
                 check_cols=self.n_features_in_
             ).to_output('array')
-            intercept_arr = CumlArray.from_input(
-                self.intercept_
-            ).to_output('array')
-            preds_arr = X_arr @ coef_arr + intercept_arr
+            if intercept_is_scalar:  # support scalar inercept_ values
+                intercept_ = self.intercept_
+            else:
+                intercept_ = CumlArray.from_input(
+                    self.intercept_,
+                    convert_to_dtype=self.dtype if isinstance(self.intercept_, float) else False
+                ).to_output('array')
+            # Ensure coefficient array has correct shape for matrix
+            # multiplication. This is a workaround specifically for the Ridge
+            # estimator where the cuml implementation of Ridge uses a 2D coef_
+            # array with shape (c, n) whereas the sklearn Ridge uses a coef_
+            # array with shape (n,c).
```

Comment:
This seems worrisome to me - I don't like that `coef_` differs between `cuml` and `sklearn` AND that in accel mode we opt to follow the sklearn convention but otherwise do it differently.

A different place where this came up we fixed our code to be compatible with sklearn so the shape of the attribute matched - I'd much prefer that fix here. Very unlikel case, but what if `n_features == n_rows` and we can't detect that a transpose needs to happen? Fixing the core problem so we don't have to add special cases and branches feels safer to me.